hid.pyx

```diff
@@ -330,6 +330,38 @@ cdef class device:
             result = hid_send_feature_report(c_hid, cbuff, c_buff_len)
         return result
 
+    def get_report_descriptor(self, int max_length=4096):
```

Comment:
`max_length` is not a parameter, but a constant in the scope of `hid_get_report_descriptor` usage

Reply:
I tested and it is possible to limit the return length of the report descriptor. 

In most case there is no interest in doing so, but I added it as an option anyway.

Reply:
Sure, you'll blindly truncate the buffer/descriptor that way, technically it will work, but there is no sane reason to do so.  By having it as an API parameter, one might think that it could make sense to limit the buffer, but by the HID Standard - it doesn't make any sense.

Reply:
@Youw so what do you suggest? Rename `max_length` to `length` or just remove it entirely? I am open to any suggestion. 

If `max_length` is removed, a buffer of length 4096 is always generated since we don't know the exact length of the report descriptor beforehand.

Reply:
> since we don't know the exact length of the report descriptor beforehand

That's the point - you never know it beforehand. You may only know it for specific known device, in which case you probably don't need the descriptor at al, as you probably know already what reports are there.

Yes, the suggestion is to remove it completely.

I want to try the cytho native array optimisation, so I might as well remove this parameter.

Reply:
I created a new PR to remove the parameter: https://github.com/trezor/cython-hidapi/pull/189